packages/kit/src/vite/index.js

```diff
@@ -61,7 +62,53 @@ const enforced_config = {
  * @return {import('vite').Plugin[]}
  */
 export function sveltekit() {
-	return [...svelte(), kit()];
+	return [...svelte(), env(), kit()];
+}
+
+/**
+ * Returns the SvelteKit environment plugin. Replaces `$env/static/*` imports
+ * with the actual environment variables.
+ * @return {import('vite').Plugin}
+ */
+function env() {
+	/** @type {import('types').ValidatedConfig} */
+	let svelte_config;
+
+	/** @type {import('vite').ConfigEnv} */
+	let vite_config_env;
+
+	return {
+		name: 'vite-plugin-svelte-kit:env',
+
+		async config(_, config_env) {
+			vite_config_env = config_env;
+			svelte_config = await load_config();
+		},
+
+		async resolveId(id) {
+			switch (id) {
+				case '$env/static/private':
+					return '\0$env/static/private';
+				case '$env/static/public':
+					return '\0$env/static/public';
+			}
+		},
+
+		async load(id) {
+			switch (id) {
+				case '\0$env/static/private':
+					return create_env_module(
+						'$env/static/private',
+						get_env(vite_config_env.mode, svelte_config.kit.env.publicPrefix).private
+					);
+				case '\0$env/static/public':
+					return create_env_module(
+						'$env/static/private',
```

Comment:
```suggestion
						'$env/static/public',
```

I guess this should be `$env/static/public`.

Reply:
Thanks! You da bomb